iaso/api/mobile/entity.py

```diff
@@ -0,0 +1,115 @@
+from django.db.models import Prefetch
+from django_filters.rest_framework import DjangoFilterBackend  # type: ignore
+from rest_framework import filters
+from rest_framework import serializers
+from rest_framework.pagination import PageNumberPagination
+
+from iaso.api.common import (
+    ModelViewSet,
+    DeletionFilterBackend,
+    TimestampField,
+)
+from iaso.models import Entity, Instance, OrgUnit, FormVersion
+
+
+class LargeResultsSetPagination(PageNumberPagination):
+    page_size = 1000
+    page_size_query_param = "page_size"
+    max_page_size = 1000
+
+
+class MobileEntityAttributesSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Instance
+        fields = ["id", "form_id", "form_version_id", "created_at", "updated_at", "org_unit_id", "json"]
+
+    form_id = serializers.IntegerField(read_only=True, source="form.id")
+    id = serializers.CharField(read_only=True, source="uuid")
+    org_unit_id = serializers.CharField(read_only=True, source="org_unit.id")
+    form_version_id = serializers.SerializerMethodField()
+
+    created_at = TimestampField()
+    updated_at = TimestampField()
+
+    @staticmethod
+    def get_form_version_id(obj: Instance):
+        form_version_id = FormVersion.objects.get(
+            version_id=obj.json.get("_version"), form__form_id=obj.form.form_id  # type: ignore
+        ).id
+
+        return form_version_id
+
+
+class MobileEntitySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Entity
+        fields = [
+            "id",
+            "created_at",
+            "updated_at",
+            "defining_instance_id",
+            "entity_type_id",
+            "instances",
+        ]
+
+    created_at = TimestampField()
+    updated_at = TimestampField()
+
+    instances = serializers.SerializerMethodField()
+    id = serializers.CharField(read_only=True, source="uuid")
+    defining_instance_id = serializers.CharField(read_only=True, source="attributes.uuid")
+    entity_type_id = serializers.CharField(read_only=True, source="entity_type.id")
+
+    @staticmethod
+    def get_instances(entity: Entity):
+        res = []
+        for instance in entity.non_deleted_instances:  # type: ignore
+            res.append(MobileEntityAttributesSerializer(instance).data)
+
+        return res
```

Comment:
Just for curiosity, wouldn't it be better to use `map`?

```suggestion
    @staticmethod
    def get_instances(entity: Entity):
        return map(
            lambda instance: MobileEntityAttributesSerializer(instance).data, 
            entity.non_deleted_instances
        )
```